projects/rocprofiler-sdk/source/lib/python/rocpd/merge.py

```diff
@@ -0,0 +1,364 @@
+    dest_path_dirname = os.path.dirname(os.path.abspath(dest_path)) or os.getcwd()
+    os.makedirs(dest_path_dirname, exist_ok=True)
+
+    with sqlite3.connect(dest_path) as conn:
+        conn.execute("PRAGMA journal_mode = WAL;")
+        conn.execute("PRAGMA synchronous = NORMAL;")
+        conn.execute("PRAGMA foreign_keys = OFF;")  # defer FK checks until end
+
+        uuids = []
+        views = []
+        # One big atomic transaction
+        with conn:
+            # Attach sources one by one
+            for i, src in enumerate(sources, 1):
+                alias = f"src{i}"
+                conn.execute(f"ATTACH DATABASE ? AS {alias}", (src,))
+                log(f"Attached {src} AS {alias}")
+
+                _uuids = [
+                    itr[0]
+                    for itr in conn.execute(
+                        f"SELECT value FROM {alias}.rocpd_metadata WHERE tag='uuid'",
+                    ).fetchall()
+                ]
+                uuids += [itr for itr in _uuids if itr not in uuids]
+
+                # Helper: fetch rows from attached sqlite_master
+                def fetch_master(_alias: str, kind: str):
+                    cur = conn.execute(
+                        f"""
+                        SELECT name, sql
+                        FROM {_alias}.sqlite_master
+                        WHERE type = ? AND name NOT LIKE 'sqlite_%'
+                        ORDER BY name
+                        """,
+                        (kind,),
+                    )
+                    return cur.fetchall()
+
+                # Track dest tables to detect collisions quickly
+                existing_tables = {
+                    row[0]
+                    for row in conn.execute(
+                        "SELECT name FROM sqlite_master WHERE type='table' AND name NOT LIKE 'sqlite_%'"
+                    )
+                }
+
+                # 1) Create tables
+                for name, create_sql in fetch_master(alias, "table"):
+                    if name in existing_tables:
+                        raise AssertionError(
+                            f"Table name collision for '{name}' from {alias}; "
+                            "assumption of globally-unique table names violated."
+                        )
+                    if not create_sql:
+                        continue  # virtual tables sometimes have NULL SQL; skip schema and rely on data?
+                    log(f"Creating table {name}")
+                    conn.execute(create_sql)
+                    existing_tables.add(name)
+
+                # 2) Copy table data
+                # Column order is identical because we just recreated the same schema text.
+                tbls = [name for name, _ in fetch_master(alias, "table")]
+                for name in tbls:
+                    log(f"Inserting rows into {name} from {alias}.{name}")
+                    conn.execute(f'INSERT INTO "{name}" SELECT * FROM {alias}."{name}"')
```

Comment:
```suggestion
                for name in tbls:
                    log(f"Inserting rows into {name} from {alias}.{name}")
                    rows = conn.execute(f'SELECT * FROM {alias}."{name}"').fetchall()
                    if rows:
                        col_count = len(rows[0])
                        placeholders = ", ".join(["?"] * col_count)
                        conn.executemany(
                            f'INSERT INTO "{name}" VALUES ({placeholders})',
                            rows
                        )
```

Looking at merge.py, @bgopesh suggested I change this to a batch operation in my PR.  Doing that in your implementation will halve the merge time!  I'm looking at ways of incorporating your merge.py with the merge.py that the Silo team wrote in my PR.  But this is a good change to pickup.